nss/src/client/mod.rs

```diff
@@ -0,0 +1,22 @@
+use authd::nss_client::NssClient;
+use std::error::Error;
+use tokio::net::UnixStream;
+use tonic::transport::{Channel, Endpoint, Uri};
+use tower::service_fn;
+
+use crate::debug;
+
+pub mod authd {
+    tonic::include_proto!("authd");
+}
+
+/// new_client creates a new client connection to the gRPC server or returns an active one.
+pub async fn new_client() -> Result<NssClient<Channel>, Box<dyn Error>> {
+    debug!("Connecting to authd on {}...", super::SOCKET_PATH);
+
+    let ch = Endpoint::try_from("http://[::]:50051")?
+        .connect_with_connector(service_fn(|_: Uri| UnixStream::connect(super::SOCKET_PATH)))
```

Comment:
As discussed during an HO, let's leave it as one connection per call to ensure that the connection gets closed and explore a way to keep the connection open for a while before closing it in an intelligent way.